src/kibali/ProtectedResource.cs

```diff
@@ -333,11 +337,20 @@ private HashSet<string> Disambiguate(string method, string scheme, HashSet<strin
         
         private (string least, string higher) ExtractScopes(IEnumerable<string> orderedScopes, HashSet<string> leastPrivilege)
         {
-            var least = leastPrivilege != null && leastPrivilege.Any() ? leastPrivilege.First() : "Not supported.";
+            var least = leastPrivilege != null && leastPrivilege.Any() ? leastPrivilege.First() : StringConstants.PermissionNotSupported;
             var filteredScopes = orderedScopes.Where(s => s!= least);
-            var higher = filteredScopes.Any() ? string.Join(", ", filteredScopes) : leastPrivilege != null && leastPrivilege.Any() ? "Not available." : "Not supported.";
+            var higher = filteredScopes.Any() ? string.Join(", ", filteredScopes) : leastPrivilege != null && leastPrivilege.Any() ? StringConstants.PermissionNotAvailable : StringConstants.PermissionNotSupported;
             return (least, higher);
         }
-    }
 
+        private bool TableRowIsValid(string least, string higher)
+        {
+            // Row is invalid if we don't have least privilege permissions but have higher privileged permissions.
+            if (least == StringConstants.PermissionNotSupported && (higher != StringConstants.PermissionNotSupported && higher != StringConstants.PermissionNotAvailable))
```

Comment:
Since we're comparing strings, we should probably pass a StringComparison parameter. https://learn.microsoft.com/en-us/dotnet/standard/base-types/best-practices-strings#recommendations-for-string-usage
```suggestion
            if (least.Equals(StringConstants.PermissionNotSupported,StringComparison.OrdinalIgnoreCase) && (!higher.Equals(StringConstants.PermissionNotSupported, StringComparison.OrdinalIgnoreCase) && !higher.Equals(StringConstants.PermissionNotAvailable, StringComparison.OrdinalIgnoreCase)))
```